pyzeebe/channel/oauth_channel.py

```diff
@@ -53,12 +71,17 @@ def create_oauth2_client_credentials_channel(
         InvalidOAuthCredentialsError: One of the provided camunda credentials is not correct
     """
 
+    authorization_server = get_camunda_oauth_url(authorization_server)
+
+    if not authorization_server:
+        raise ValueError("ZEEBE_AUTHORIZATION_SERVER_URL is not configured")
+
     oauth2_client_credentials = Oauth2ClientCredentialsMetadataPlugin(
-        client_id=client_id,
-        client_secret=client_secret,
-        authorization_server=authorization_server,
-        scope=scope,
-        audience=audience,
+        client_id=get_camunda_client_id(client_id),
+        client_secret=get_camunda_client_secret(client_secret),
+        authorization_server=authorization_server or "",
```

Comment:
I don't get why `or ""`. 😅

Reply:
Just to make the type checker happy :sweat_smile: 
Otherwise the type would be `str | None`, but the parameter type is just `str`

Reply:
Ah okay, but since it's not optional it should be set which means woever sets this parameter should not assign `None` (This would mean the `get_camunda_oauth_url`)

Reply:
@felicijus right, if this parameter is missing for some reason, a `ValueError` is already raised in line 77. Sadly the type checker does not consider this

Reply:
Should the error not be rased in the `get_camunda_oauth_url` function?
Return type should be str not optional[str]

Also could you change to [PEP604](https://peps.python.org/pep-0604/) recent change from @dimastbk.

Reply:
The problem is that in case of `create_camunda_cloud_channel()` it needs to be possible to use `"https://login.cloud.camunda.io/oauth/token"` as additional fallback. So `get_camunda_oauth_url` should not throw there.

Reply:
Is the [PEP604](https://peps.python.org/pep-0604/) change already in master?

Reply:
Yes. You may need to sync your branch.

Reply:
I have rebased the branch and applied PEP604

Reply:
Can you revert the changes to the partial too ?

Reply:
Done, see https://github.com/camunda-community-hub/pyzeebe/pull/509/commits/12535fc372f7bd387be09deff82ae8977d6da396